pkg/oam/util/helper.go

```diff
@@ -143,3 +175,24 @@ func Object2Map(obj interface{}) (map[string]interface{}, error) {
 	err = json.Unmarshal(bts, &res)
 	return res, err
 }
+
+// LocateReplicaField call openapi RESTFUL end point to fetch the schema of the resource and try to see
+// 					 if it has a spec.replicas filed that is of type integer
+func LocateReplicaField(document openapi.Resources, res *unstructured.Unstructured) bool {
+	replicaFieldPath := []string{"spec", "replicas"}
```

Comment:
How can you make sure every CRD has replica mechanism using  `spec.replicas`? I think it's possible for any names(eg `spec.replicaCount`, `spec.replica`). What we need is a mapping table that can allow trait builder to fill into traitdefinition

Reply:
It's fine to make this assumption for now. We should be cautious on making definition object a "black magic". 

@ryanzhang-oss Could you please add comment or TODO to clarify this assumption.

Reply:
@wonderflow This is by no means to cover all cases. It is just an extra check on top of the duck typing we already utilized. The alternative is to list all the resource GVKs (deployment/statefulset/cloneset...). I found the openAPI route less clumsy and more extensible. Granted, it won't solve the cases you listed. We can add field path mapping somewhere but IMO, it seems to be an overkill for now.
@resouer  This is still WIP so yes, I will add tests and comments on this.